AppServices/CustomerAppService.cs

```diff
@@ -1,41 +1,56 @@
-﻿using APIDesafioWarren.DataBase;
+﻿using APIDesafioWarren.DomainService;
 using APIDesafioWarren.Models;
+using App.Services;
+using Application.Models.DTOs;
+using AutoMapper;
 using System;
 using System.Collections.Generic;
 
 namespace AppServices
 {
     public class CustomerAppService : ICustomerAppService
     {
-        private readonly ICustomerServices _customerServices;
-        public CustomerAppService(ICustomerServices customerServices)
+        private readonly ICustomerServices _customerService;
+
+        private readonly IMapper _mapper;
+
+        public CustomerAppService(ICustomerServices customerService, IMapper mapper)
         {
-            _customerServices = customerServices;
+            _customerService = customerService ?? throw new ArgumentNullException(nameof(customerService));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
         }
 
-        public List<Customer> GetAll(Predicate<Customer> predicate = null)
+        public IEnumerable<CustomerResponse> GetAll(Predicate<Customer> predicate = null)
         {
-            return _customerServices.GetAll(predicate);
+            var getAllCustomers = _customerService.GetAll(predicate);
+            return _mapper.Map<IEnumerable<CustomerResponse>>(getAllCustomers);
         }
 
         public Customer GetBy(Predicate<Customer> predicate)
```

Comment:
```suggestion
        public CustomerResponse GetBy(Predicate<Customer> predicate)
```
Devemos retornar a DTO `CustomerResponse`